### PR TITLE
feat(curriculum): add css-typography quiz

### LINKED ISSUE #56188: [HACKTOBERFEST 2024] - CSS Typography Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the CSS Typography quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-css-typography`.

**All quiz questions need to relate to one of these topics:**

- fundamentals of typography
- best practises
- font-family
- web safe fonts
- `@font-face` rule
- external fonts (e.g. google fonts)
- text-shadow property

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by Dhairya-A-Mehra**
Checklist:

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or GitPod.

Closes #56188 

I have added questions for the CSS Typography Quiz. I have not tested this application on my local machine because I am developing this on Windows. However, I consulted with @naomi-lgbt , and they informed me that I am allowed to submit the PR without local testing.
If any changes to the questions are needed, I am happy to make adjustments as required.

### CODE REVIEW

**Comment by ilenia-magoni** [comment]
request: do not keep closing and opening PRs, it creates a lot of unnecessary noise

**Comment by Dhairya-A-Mehra** [comment]
> request: do not keep closing and opening PRs, it creates a lot of unnecessary noise

do you want me to close this PR?

**Comment by ilenia-magoni** [comment]
no, I just asked you to stop closing and reopening, keep this one and don't close and reopen again please

**Comment by Dhairya-A-Mehra** [comment]
> no, I just asked you to stop closing and reopening, keep this one and don't close and reopen again please
okay sure
last one I closed because it was not in proper format given by you

**Comment by ilenia-magoni** [comment]
you can edit the title instead of closing and reopening a PR

**Comment by ilenia-magoni** [comment]
You also do not need to make a new PR if you are asked to change something in the code, there you will need to push a new commit

File: curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md [hunk 1]
```diff
@@ -17,439 +17,416 @@ Answer all of the questions below correctly to pass the quiz.
 
 #### --text--
 
-Placeholder question
+Which of the following is the correct way to write a **@font-face** rule in CSS?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md:20]:
```suggestion
Which of the following is the correct way to write a `@font-face` rule in CSS?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md [hunk 2]
```diff
@@ -17,439 +17,416 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --distractors--
 
-Placeholder distractor 1
+@font-face { font: 'MyFont'; size: 12px; }
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md:24]:
```suggestion
`@font-face { font: 'MyFont'; size: 12px; }`
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Before I can review this in-depth, all code needs to be between backticks so our parser renders it in code tags. I've left a couple of examples.

**Comment by Dhairya-A-Mehra** [comment]
> Before I can review this in-depth, all code needs to be between backticks so our parser renders it in code tags. I've left a couple of examples.

Okay I'll do it

File: curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md [hunk 3]
```diff
@@ -17,439 +17,416 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+3 or 4
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What CSS property is used to set the **font family** for an element?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md:240]:
```suggestion
What CSS property is used to set the font family for an element?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md [hunk 4]
```diff
@@ -17,439 +17,416 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`font-family`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Why is **typographic hierarchy** important in design?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md:262]:
```suggestion
Why is typographic hierarchy important in design?
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Comment by ilenia-magoni** [comment]
note, you can make a single commit if you go to the Edited files tab, which is preferred

**Comment by Dhairya-A-Mehra** [comment]
> note, you can make a single commit if you go to the Edited files tab, which is preferred

I tried to commit this thing as a batch, but I was not able to do that, so made single commits.
Sorry for that

File: curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md [hunk 5]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+It changes the text color
 
 #### --answer--
 
-Placeholder answer
-
+It controls the amount of space between lines of text
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md:454]:
```suggestion
It controls the amount of space between lines of text

```

you accidentally removed a line here, let's put it back

**Review by naomi-lgbt — APPROVED** [review]

File: curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md [hunk 6]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+It ensures a consistent experience in case the custom font doesn’t load
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How many values does the `text-shadow` property usually have?
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md:218]: I don't quite like this question. Traditionally speaking, yes, developers probably want more than just the two required offset values but it feels unverifiable.

File: curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md [hunk 7]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+3 or 4
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What CSS property is used to set the font family for an element?
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md:240]: This is a give-me question. You said the answer in the question.  Please change it.

File: curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md [hunk 8]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`text-shadow: 2px 2px 5px rgba(0, 0, 0, 0.5);`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does the first value in the `text-shadow` property control?
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md:328]: Actually the first value, while optional, controls the color of the shadow. 

Source: https://developer.mozilla.org/en-US/docs/Web/CSS/text-shadow

File: curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md [hunk 9]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+The horizontal offset of the shadow
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+For body text, what type of font is generally the easiest to read?
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/quiz-css-typography/66ed9010f45ce3ece4053eb8.md:350]: What exactly makes a serif font less readable than a sans-serif font? After all Times New Roman is pretty legible.

**Review by a2937 — CHANGES_REQUESTED** [review]

**Comment by Dhairya-A-Mehra** [comment]
@a2937 I'll do the require changes

**Comment by gikf** [comment]
Hey @Dhairya-A-Mehra,

Any progress on addressing requested changes?

**Comment by Dhairya-A-Mehra** [comment]
> Hey @Dhairya-A-Mehra,
> 
> Any progress on addressing requested changes?

I will be committing the required changes in a few hours.

**Review by a2937 — CHANGES_REQUESTED** [review]
You still have yet to respond to Ilenia's suggestion.

**Comment by Dhairya-A-Mehra** [comment]
> You still have yet to respond to Ilenia's suggestion.

Oh, I am extremely sorry.

**Review by a2937 — APPROVED** [review]
Everything seems okay to me.

**Comment by gikf** [comment]
Hi @Dhairya-A-Mehra. Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Comment by Dhairya-A-Mehra** [comment]
> Hi @Dhairya-A-Mehra. Congrats on your first pull request (PR)! 🎉
> 
> Thank you for your contribution to the page! 👍 We are happy to accept these changes and look forward to future contributions. 📝

Thank You all for supporting me through this journey of contributing in your organization.

**Pull request merged**